okhttp/src/main/java/okhttp3/internal/ws/RealWebSocket.kt

```diff
@@ -162,15 +169,24 @@ class RealWebSocket(
         try {
           checkUpgradeSuccess(response, exchange)
           streams = exchange!!.newWebSocketStreams()
-          // TODO(jwilson): use request & response headers to negotiate extensions.
-          extensions = WebSocketExtensions()
         } catch (e: IOException) {
           exchange?.webSocketUpgradeFailed()
           failWebSocket(e, response)
           response.closeQuietly()
           return
         }
 
+        // Apply the extensions. If they're unacceptable initiate a graceful shut down.
+        // TODO(jwilson): Listeners should get onFailure() instead of onClosing() + onClosed(1010).
```

Comment:
@swankjesse  
Is there any reason not to use failWebSocket() here and return?
If not I am happy to submit a pr to fix this TODO